[HttpBase] ApacheRequestHeadersBuilder fails when header value is null

__Describe the Bug__

A customer that sent an illegal header value ~~(`{Accept: null}`)~~ (`{Content-Type: null}`) saw errors when after we upgraded the Cloudproxy function to the latest HTTP client (8.6.0-alpha3).

Reason: The `ApacheRequestHeadersBuilder` fails if a header entry value is `null` when calling `defaultMultipartContentType()` as there is no null check before accessing the header entry value.

__Steps to Reproduce__

The customer sent the following payload for the `headers` from a 8.3 Connectors runtime: 
```
{Content-Type: application/json}
```

and forgot to quote the `application/json` which is a valid variable name in FEEL leading to the following value for the `headers`:
```
{Content-Type: null}
```

__Expected Behavior__

The header value should be checked for a null value before accessing the entry value:
```
e.getValue().contains(MULTIPART_FORM_DATA.getMimeType())
```

__Environment__
- Camunda version: 8.3 with Cloud Proxy Function `8.6.0-alpha3`

https://jira.camunda.com/browse/SUPPORT-22639

Stack Trace from [the logs](https://cloudlogging.app.goo.gl/j9UgV6q579ZFUETJ8):

```java

Exception while processing job: 2251799872426124 for tenant: <default>
io.camunda.connector.api.error.ConnectorException: java.lang.NullPointerException: Cannot invoke "String.contains(java.lang.CharSequence)" because the return value of "java.util.Map$Entry.getValue()" is null
	at io.camunda.connector.http.base.services.HttpInteractionService.executeHttpRequest(HttpInteractionService.java:77)
	at io.camunda.connector.http.base.services.HttpService.executeRequestViaProxy(HttpService.java:103)
	at io.camunda.connector.http.base.services.HttpService.executeConnectorRequest(HttpService.java:53)
	at io.camunda.connector.http.rest.HttpJsonFunction.execute(HttpJsonFunction.java:79)
	at io.camunda.connector.runtime.core.outbound.ConnectorJobHandler.handle(ConnectorJobHandler.java:105)
	at io.camunda.connector.runtime.outbound.jobhandling.SpringConnectorJobHandler.lambda$handle$0(SpringConnectorJobHandler.java:75)
	at io.micrometer.core.instrument.AbstractTimer.record(AbstractTimer.java:247)
	at io.camunda.zeebe.spring.client.actuator.MicrometerMetricsRecorder.executeWithTimer(MicrometerMetricsRecorder.java:52)
	at io.camunda.connector.runtime.outbound.jobhandling.SpringConnectorJobHandler.handle(SpringConnectorJobHandler.java:66)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactoryImpl.executeJob(JobRunnableFactoryImpl.java:45)
	at io.camunda.zeebe.client.impl.worker.JobRunnableFactoryImpl.lambda$create$0(JobRunnableFactoryImpl.java:40)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: com.google.api.client.http.HttpResponseException: 500 Internal Server Error
POST https://europe-west1-camunda-saas-prod-connectors.cloudfunctions.net/connector-http-json-proxy
{"error":"java.lang.NullPointerException: Cannot invoke \"String.contains(java.lang.CharSequence)\" because the return value of \"java.util.Map$Entry.getValue()\" is null","errorCode":"Failed to execute request: HttpRequest{url='https://oa-introducer-search.azurewebsites.net/api/ClientCompany/Search', method='POST', authentication=OAuthAuthentication{oauthTokenEndpoint='https://xxxx', clientId='xxxxx', clientSecret=[REDACTED], audience='https://introducerqueryapi.axsesstoday.com.au', clientAuthentication='basicAuthHeader', scopes='null'}, connectionTimeoutInSeconds='20', headers={Content-Type=null}', readTimeoutInSeconds='null, body={filter=clientCompanyId eq 'xxx'}, queryParameters=null}. An error occurred: Cannot invoke \"String.contains(java.lang.CharSequence)\" because the return value of \"java.util.Map$Entry.getValue()\" is null","result":null}
	at com.google.api.client.http.HttpResponseException$Builder.build(HttpResponseException.java:293)
	at com.google.api.client.http.HttpRequest.execute(HttpRequest.java:1118)
	at io.camunda.connector.http.base.services.HttpInteractionService.executeHttpRequest(HttpInteractionService.java:63)
	... 16 common frames omitted

```

--- a/connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/builder/parts/ApacheRequestHeadersBuilder.java
+++ b/connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/builder/parts/ApacheRequestHeadersBuilder.java
@@ -31,6 +31,8 @@ public class ApacheRequestHeadersBuilder implements ApacheRequestPartBuilder {
 
   @Override
   public void build(ClassicRequestBuilder builder, HttpCommonRequest request) {
+    sanitizeContentType(request);
+
     var hasContentTypeHeader =
         Optional.ofNullable(request.getHeaders())
             .map(headers -> headers.containsKey(CONTENT_TYPE))
@@ -59,5 +61,12 @@ public class ApacheRequestHeadersBuilder implements ApacheRequestPartBuilder {
         e.getKey().equals(CONTENT_TYPE)
             && e.getValue().contains(MULTIPART_FORM_DATA.getMimeType())
             && !e.getValue().contains("boundary");
+  }
+
+  /** Remove the content type header if it is {@code null}. */
+  private void sanitizeContentType(HttpCommonRequest request) {
+    if (request.getHeaders() != null && request.getHeaders().get(CONTENT_TYPE) == null) {
+      request.getHeaders().remove(CONTENT_TYPE);
+    }
   }
 }

--- a/connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/builder/parts/ApacheRequestHeadersBuilder.java
+++ b/connectors/http/http-base/src/main/java/io/camunda/connector/http/base/client/apache/builder/parts/ApacheRequestHeadersBuilder.java
@@ -22,6 +22,7 @@ import static org.apache.hc.core5.http.ContentType.MULTIPART_FORM_DATA;
 import static org.apache.hc.core5.http.HttpHeaders.CONTENT_TYPE;
 
 import io.camunda.connector.http.base.model.HttpCommonRequest;
+import java.util.HashMap;
 import java.util.Map;
 import java.util.Optional;
 import java.util.function.Predicate;
@@ -31,20 +32,15 @@ public class ApacheRequestHeadersBuilder implements ApacheRequestPartBuilder {
 
   @Override
   public void build(ClassicRequestBuilder builder, HttpCommonRequest request) {
-    sanitizeContentType(request);
+    var headers = sanitizedHeaders(request);
 
     var hasContentTypeHeader =
-        Optional.ofNullable(request.getHeaders())
-            .map(headers -> headers.containsKey(CONTENT_TYPE))
-            .orElse(false);
+        Optional.of(headers).map(hd -> hd.containsKey(CONTENT_TYPE)).orElse(false);
     if (request.getMethod().supportsBody && !hasContentTypeHeader) {
       // default content type
       builder.addHeader(CONTENT_TYPE, APPLICATION_JSON.getMimeType());
     }
-    if (request.getHeaders() == null) {
-      request.setHeaders(new java.util.HashMap<>());
-    }
-    request.getHeaders().entrySet().stream()
+    headers.entrySet().stream()
         .filter(not(defaultMultipartContentType()))
         .forEach(e -> builder.addHeader(e.getKey(), e.getValue()));
   }
@@ -64,9 +60,12 @@ public class ApacheRequestHeadersBuilder implements ApacheRequestPartBuilder {
   }
 
   /** Remove the content type header if it is {@code null}. */
-  private void sanitizeContentType(HttpCommonRequest request) {
-    if (request.getHeaders() != null && request.getHeaders().get(CONTENT_TYPE) == null) {
-      request.getHeaders().remove(CONTENT_TYPE);
+  private Map<String, String> sanitizedHeaders(HttpCommonRequest request) {
+    var headers =
+        Optional.ofNullable(request.getHeaders()).map(HashMap::new).orElse(new HashMap<>());
+    if (headers.get(CONTENT_TYPE) == null) {
+      headers.remove(CONTENT_TYPE);
     }
+    return headers;
   }
 }

--- a/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/HttpServiceTest.java
+++ b/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/HttpServiceTest.java
@@ -234,6 +234,7 @@ public class HttpServiceTest {
     request.setBody(Map.of("name", "John", "age", 30, "message", "{\"key\":\"value\"}"));
     Map<String, String> headers = new HashMap<>();
     headers.put("Content-Type", null);
+    headers.put("Other", null);
     request.setHeaders(headers);
     request.setUrl(getHostAndPort(wmRuntimeInfo) + "/path");
 
@@ -250,6 +251,7 @@ public class HttpServiceTest {
     verify(
         postRequestedFor(urlEqualTo("/path"))
             .withHeader("Content-Type", equalTo("application/json"))
+            .withHeader("Other", equalTo(""))
             .withRequestBody(
                 matchingJsonPath("$.name", equalTo("John"))
                     .and(matchingJsonPath("$.message", equalTo("{\"key\":\"value\"}")))

--- a/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/HttpServiceTest.java
+++ b/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/HttpServiceTest.java
@@ -45,6 +45,7 @@ import io.camunda.connector.http.base.cloudfunction.CloudFunctionService;
 import io.camunda.connector.http.base.model.HttpCommonRequest;
 import io.camunda.connector.http.base.model.HttpCommonResult;
 import io.camunda.connector.http.base.model.HttpMethod;
+import java.util.HashMap;
 import java.util.Map;
 import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.params.ParameterizedTest;
@@ -202,6 +203,57 @@ public class HttpServiceTest {
     assertThat(e).isNotNull();
     assertThat(e.getErrorCode()).isEqualTo("401");
     assertThat(e.getMessage()).isEqualTo("Unauthorized sorry!");
+    if (cloudFunctionEnabled) {
+      verify(
+          postRequestedFor(urlEqualTo("/proxy"))
+              .withRequestBody(equalTo(objectMapper.writeValueAsString(request))));
+    }
+  }
+
+  @ParameterizedTest
+  @ValueSource(booleans = {true, false})
+  public void shouldReturn200WithBody_whenPostRequestWithNullContentType(
+      boolean cloudFunctionEnabled, WireMockRuntimeInfo wmRuntimeInfo) throws Exception {
+    stubCloudFunction(wmRuntimeInfo);
+    HttpService httpService =
+        cloudFunctionEnabled ? HttpServiceTest.this.httpService : httpServiceWithoutCloudFunction;
+
+    stubFor(
+        post("/path")
+            .willReturn(
+                ok().withJsonBody(
+                        JsonNodeFactory.instance
+                            .objectNode()
+                            .put("responseKey1", "value1")
+                            .put("responseKey2", 40)
+                            .putNull("responseKey3"))));
+
+    // given
+    HttpCommonRequest request = new HttpCommonRequest();
+    request.setMethod(HttpMethod.POST);
+    request.setBody(Map.of("name", "John", "age", 30, "message", "{\"key\":\"value\"}"));
+    Map<String, String> headers = new HashMap<>();
+    headers.put("Content-Type", null);
+    request.setHeaders(headers);
+    request.setUrl(getHostAndPort(wmRuntimeInfo) + "/path");
+
+    // when
+    HttpCommonResult result = httpService.executeConnectorRequest(request);
+
+    // then
+    assertThat(result).isNotNull();
+    assertThat(result.status()).isEqualTo(200);
+    JSONAssert.assertEquals(
+        "{\"responseKey1\":\"value1\",\"responseKey2\":40,\"responseKey3\":null}",
+        objectMapper.writeValueAsString(result.body()),
+        JSONCompareMode.STRICT);
+    verify(
+        postRequestedFor(urlEqualTo("/path"))
+            .withHeader("Content-Type", equalTo("application/json"))
+            .withRequestBody(
+                matchingJsonPath("$.name", equalTo("John"))
+                    .and(matchingJsonPath("$.message", equalTo("{\"key\":\"value\"}")))
+                    .and(matchingJsonPath("$.age", equalTo("30")))));
     if (cloudFunctionEnabled) {
       verify(
           postRequestedFor(urlEqualTo("/proxy"))

--- a/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/client/apache/ApacheRequestFactoryTest.java
+++ b/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/client/apache/ApacheRequestFactoryTest.java
@@ -36,13 +36,17 @@ import io.camunda.connector.http.base.model.auth.BasicAuthentication;
 import io.camunda.connector.http.base.model.auth.BearerAuthentication;
 import io.camunda.connector.http.base.model.auth.OAuthAuthentication;
 import java.nio.charset.StandardCharsets;
+import java.util.HashMap;
 import java.util.Map;
 import org.apache.hc.core5.http.ClassicHttpRequest;
 import org.apache.hc.core5.http.ContentType;
 import org.apache.hc.core5.http.Header;
 import org.apache.hc.core5.http.HttpHeaders;
+import org.apache.hc.core5.http.ProtocolException;
 import org.junit.jupiter.api.Nested;
 import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.EnumSource;
 import org.mockito.MockedStatic;
 
 public class ApacheRequestFactoryTest {
@@ -410,6 +414,31 @@ public class ApacheRequestFactoryTest {
 
   @Nested
   class HeadersTests {
+
+    @ParameterizedTest
+    @EnumSource(HttpMethod.class)
+    public void shouldSetContentType_whenNullProvidedAndPost(HttpMethod method)
+        throws ProtocolException {
+      // given request without content type
+      HttpCommonRequest request = new HttpCommonRequest();
+      request.setMethod(method);
+      Map<String, String> headers = new HashMap<>();
+      headers.put(HttpHeaders.CONTENT_TYPE, null);
+      headers.put(HttpHeaders.ACCEPT, null);
+      headers.put("Other", null);
+      request.setHeaders(headers);
+
+      // when
+      ClassicHttpRequest httpRequest = ApacheRequestFactory.get().createHttpRequest(request);
+
+      // then
+      if (method.supportsBody) {
+        assertThat(httpRequest.getHeader(HttpHeaders.CONTENT_TYPE).getValue())
+            .isEqualTo(ContentType.APPLICATION_JSON.getMimeType());
+      } else {
+        assertThat(httpRequest.getHeader(HttpHeaders.CONTENT_TYPE)).isNull();
+      }
+    }
 
     @Test
     public void shouldSetJsonContentType_WhenNotProvidedAndSupportsBody() throws Exception {

--- a/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/client/apache/CustomApacheHttpClientTest.java
+++ b/connectors/http/http-base/src/test/java/io/camunda/connector/http/base/client/apache/CustomApacheHttpClientTest.java
@@ -63,6 +63,7 @@ import io.camunda.connector.http.base.model.auth.BasicAuthentication;
 import io.camunda.connector.http.base.model.auth.BearerAuthentication;
 import io.camunda.connector.http.base.model.auth.OAuthAuthentication;
 import java.net.URISyntaxException;
+import java.util.HashMap;
 import java.util.Map;
 import org.apache.commons.text.StringEscapeUtils;
 import org.apache.hc.core5.http.ContentType;
@@ -199,6 +200,20 @@ public class CustomApacheHttpClientTest {
 
   @Nested
   class GetTests {
+
+    @Test
+    public void shouldReturn200_whenNullHeaders(WireMockRuntimeInfo wmRuntimeInfo) {
+      stubFor(get("/path").willReturn(ok()));
+      HttpCommonRequest request = new HttpCommonRequest();
+      request.setMethod(HttpMethod.GET);
+      var headers = new HashMap<String, String>();
+      headers.put("Content-Type", null);
+      request.setHeaders(headers);
+      request.setUrl(getHostAndPort(wmRuntimeInfo) + "/path");
+      HttpCommonResult result = customApacheHttpClient.execute(request);
+      assertThat(result).isNotNull();
+      assertThat(result.status()).isEqualTo(200);
+    }
 
     @Test
     public void shouldReturn200_whenNoTimeouts(WireMockRuntimeInfo wmRuntimeInfo) {

